File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofModels/MLBTrendsRecords.swift
import Foundation

public struct MLBTrendsSlateBundle: Sendable {
    public let games: [OutliersTrendsGame]
    public let season: Int
    public let throughDate: String?
    public let teams: [MLBTeamTrendRecord]

    public init(
        games: [OutliersTrendsGame],
        season: Int,
        throughDate: String?,
        teams: [MLBTeamTrendRecord]
    ) {
        self.games = games
        self.season = season
        self.throughDate = throughDate
        self.teams = teams
    }
}

public struct MLBTeamTrendRecord: Sendable, Identifiable {
    public var id: String { teamAbbr }
    public let teamAbbr: String
    public let teamName: String?
    public let season: Int
    public let throughDate: String?
    public let splits: NFLTrendSplits
    public let matchups: [String: NFLTrendMatchupRecord]

    public init(
        teamAbbr: String,
        teamName: String?,
        season: Int,
        throughDate: String?,
        splits: NFLTrendSplits,
        matchups: [String: NFLTrendMatchupRecord]
    ) {
        self.teamAbbr = teamAbbr
        self.teamName = teamName
        self.season = season
        self.throughDate = throughDate
        self.splits = splits
        self.matchups = matchups
    }
}

File: wagerproof-android-native/core/models/src/main/kotlin/com/wagerproof/core/models/MLBTrendsRecords.kt
package com.wagerproof.core.models

/**
 * MLB team-trends slate bundle. Port of iOS `MLBTrendsRecords.swift` —
 * client-only (assembled by the trends store, never decoded directly).
 * Reuses the NFL trend-split primitives: [NFLTrendSplits] (typealias in
 * OutliersTrends.kt) and [NFLTrendMatchupRecord] (NFLTrendsRecords.kt).
 */
data class MLBTrendsSlateBundle(
    val games: List<OutliersTrendsGame>,
    val season: Int,
    val throughDate: String?,
    val teams: List<MLBTeamTrendRecord>,
)

data class MLBTeamTrendRecord(
    val teamAbbr: String,
    val teamName: String?,
    val season: Int,
    val throughDate: String?,
    val splits: NFLTrendSplits,
    val matchups: Map<String, NFLTrendMatchupRecord>,
) {
    val id: String get() = teamAbbr
}
